.github/workflows/ci.yml

```diff
@@ -362,39 +362,30 @@ jobs:
 
 
   self-hosted-end-to-end:
-    needs: []
     runs-on: ubuntu-latest
     # temporary, remove once we are confident the action is working
     continue-on-error: true
-
+    timeout-minutes: 30
     steps:
-      # TODO: move image build to its own step once we can push them
-      - name: Checkout Relay
-        uses: actions/checkout@v3
-        with:
-          submodules: recursive
-      - name: Try pulling Relay images
-        shell: bash
-        # We don't need these but they might make things faster
-        continue-on-error: true
-        run: |
-          docker pull us.gcr.io/getsentry/relay:nightly
-          docker pull us.gcr.io/getsentry/relay:deps
-      - name: Build Relay first stage
-        run: |
-          docker build --target relay-deps -t ghcr.io/getsentry/relay:deps \
-            --cache-from us.gcr.io/getsentry/relay:deps \
-            .
-      - name: Build Relay final image
+      - name: Pull the test image
+        id: image_pull
+        env:
+          RELAY_TEST_IMAGE: us.gcr.io/sentryio/relay:${{ github.event.pull_request.head.sha || github.sha }} 
         run: |
-          docker build -t ghcr.io/getsentry/relay:nightly -t ghcr.io/getsentry/relay:${{ github.sha }} \
-            --cache-from us.gcr.io/$PROJECT_ID/relay:deps \
-            --cache-from us.gcr.io/$PROJECT_ID/relay:nightly \
-            .
+          echo "We poll for the Relay Docker image that the GCB build produces until it succeeds or this job times out."
+          if [[ -z "$RELAY_TEST_IMAGE" ]]; then
+              echo "The RELAY_TEST_IMAGE needs to be set" 1>&2
+              exit 1
+          fi
+          echo "Polling for $RELAY_TEST_IMAGE"
+          until docker pull "$RELAY_TEST_IMAGE" 2>/dev/null; do
+              sleep 10
+          done
+          echo "relay-test-image-name=$RELAY_TEST_IMAGE" >> "$GITHUB_OUTPUT"
       # TODO: push the image here
       - name: Run Sentry self-hosted e2e CI
-        uses: getsentry/action-self-hosted-e2e-tests@5bbd3d6725bbe4bf5cc12f8a3291ce87df6a891f
+        uses: getsentry/action-self-hosted-e2e-tests@fa5b8240848f0e645ac2918c530e60ec8f50e4b8
```

Comment:
Sidecar version bump? Or necessary for this to work?

Reply:
Remind me, did we try using a branch name here and were asked to SHA pin?

Reply:
for `sentry` / `getsentry` we've standardized on freezing these -- but I don't have super strong opinions elsewhere